src/analyze.cpp

```diff
@@ -6124,10 +6129,12 @@ void render_const_value(CodeGen *g, Buf *buf, ConstExprValue *const_val) {
             }
         case ZigTypeIdPointer:
             return render_const_val_ptr(g, buf, const_val, type_entry);
-        case ZigTypeIdVector:
-            return render_const_val_array(g, buf, const_val, type_entry->data.vector.len);
-        case ZigTypeIdArray:
-            return render_const_val_array(g, buf, const_val, type_entry->data.array.len);
+        case ZigTypeIdArray: // fallthrough
```

Comment:
I'm ok with this fallthrough, but we would need to go change the types in all_types.hpp of the union, because one of them is accessing `type_entry->data.array.len` and one of them is accessing `type_entry->data.vector.len`. I think the type of len would need to be audited too - uint64_t vs uint32_t.

Reply:
Oops, that was certainly not intended. I believe I just did a sloppy job during a rebase; I'll check my diffs more carefully.

I believe type-punning here would be unnecessarily confusing/complex, and refactoring types is out of the scope of this patch anyways, so I've just undone the fallthrough.